analyzers/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -205,9 +215,13 @@ private Location CalculateExtendedLocation(SyntaxNode parent, bool isLeftSide)
             return isLeftSide ? parent.CreateLocation(token) : token.CreateLocation(parent);
         }
 
-        private static bool CheckForNullability(SyntaxNode left, SyntaxNode right, SemanticModel model) =>
-            right is null // Avoids DeclarationPattern or RecursivePattern
-            || model.GetTypeInfo(left).Type.IsNullableBoolean()
-            || model.GetTypeInfo(right).Type.IsNullableBoolean();
+        private bool TypeShouldBeIgnored(SyntaxNode node, SemanticModel model)
+        {
+            var type = model.GetTypeInfo(node).Type;
+            return type.IsNullableBoolean()
+                || type is ITypeParameterSymbol
+                || type.Is(KnownType.System_Object)
+                || systemBooleanInterfaces.Any(x => x.Equals(type));
```

Comment:
FYI: Regarding symbol "Equals" there is some fun coming up, when we upgrade Roslyn:
https://github.com/dotnet/roslyn-analyzers/issues/3427
Symbol equality comparison should be explicit about nullable annotation handling. Our version of Roslyn does not contain `SymbolEqualityComparer`.